build-logic/src/main/kotlin/kotlinx/io/conventions/multiplatform-lib-conventions.gradle.kts

```diff
@@ -100,6 +106,41 @@ fun NamedDomainObjectContainer<KotlinSourceSet>.createSourceSet(
 }
 
 kotlin {
+    jvmToolchain {
+        // Version catalog cannot be accessed from convention plugin. Until this problem is solved we configure the toolchain by project extension
+        // see https://github.com/gradle/gradle/issues/17963
+        languageVersion.set(extension.javaVersion)
```

Comment:
just FYI, there is possibility to do this even now using type unsafe API: https://docs.gradle.org/current/userguide/platforms.html#sub:type-unsafe-access-to-catalog

Reply:
Great, thanks!